conf/imx_hab4/sign-file.sh

```diff
@@ -0,0 +1,215 @@
+#!/bin/sh
+#
+# Copyright (c) 2019 Foundries.io
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+CST_BINARY="cst"
+CSF_TEMPLATE="u-boot-spl-sign.csf-template"
+KEY_DIR="."
+DCD_CLEAR=0
+DISPLAY_USAGE=0
+SIGN_SPL=0
+SIGN_M4APP=0
+
+function parse_args()
+{
+    while [ ${#} -gt 0 ]
+    do
+        case ${1} in
+        --cst)
+            CST_BINARY=${2}
+            shift
+            shift
+            ;;
+        --csf-template)
+            CSF_TEMPLATE=${2}
+            shift
+            shift
+            ;;
+        --spl)
+            WORK_FILE=${2}
+            SIGN_SPL=1
+            shift
+            shift
+            ;;
+        --m4app)
+            WORK_FILE=${2}
+            SIGN_M4APP=1
+            shift
+            shift
+            ;;
+        --key-dir)
+            KEY_DIR=${2}
+            shift
+            shift
+            ;;
+        --fix-sdp-dcd)
+            DCD_CLEAR=1
+            shift
+            ;;
+        -h)
+            DISPLAY_USAGE=1
+            shift
+            ;;
+        --help)
+            DISPLAY_USAGE=1
+            shift
+            ;;
+        *)
+            shift
+            ;;
+        esac
+    done
+}
+
+parse_args "$@"
+
+if [ "${DISPLAY_USAGE}" == "1" ]; then
+    echo "usage for: ${0}"
+    echo "  --cst: set cst binary path/filename   [default: ${CST_BINARY}]"
+    echo "  --csf-template: set CSF template file [default: ${CSF_TEMPLATE}]"
+    echo "  --spl: SPL binary to sign             [required]"
+    echo "  --m4app: M4 binary to sign            [required]"
```

Comment:
Is m4app required or optional?

Reply:
Has to have either --spl or --m4app to specify binary

Reply:
Maybe change usage to say one or other required? Both with [required] is confusing.